common/src/main/java/io/crate/types/IpType.java

```diff
@@ -62,15 +66,24 @@ private static String longToIp(long longIp) {
     }
 
     private void validate(String ip) {
-        if (!InetAddresses.isInetAddress(ip)) {
+        Runnable throwIAE = () -> {
             throw new IllegalArgumentException(
                 "Failed to validate ip [" + ip + "], not a valid ipv4 address");
+        };
+
+        if (ip.contains("/")) {
+            try {
+                InetAddresses.parseCidr(ip);
+            } catch (Throwable e) {
+                throwIAE.run();
+            }
+        } else if (!InetAddresses.isInetAddress(ip)) {
+            throwIAE.run();
```

Comment:
Are the changes here still required? I think these should be reverted?

Reply:
I believe they are, I will however take a closer look

Reply:
> you cannot insert if the column is declared of type ip

Why would you want to insert a CIDR into a ip column?
I think I'm missing something here.